modules/integration/tests-common/admin-clients/src/main/java/org/wso2/mb/integration/common/clients/AndesClient.java

```diff
@@ -1,589 +1,317 @@
+     * @throws IOException
      */
-    public int getTotalNumberOfDuplicates() {
-        AndesClientOutputParser andesClientOutputParser = new AndesClientOutputParser(filePathToWriteReceivedMessages);
-        return andesClientOutputParser.numberDuplicatedMessages();
+    public boolean checkIfMessagesAreInOrder()
+            throws IOException {
+        if (0 < consumers.size()) {
+            AndesClientOutputParser andesClientOutputParser =
+                    new AndesClientOutputParser(consumers.get(0).getConfig()
+                                                        .getFilePathToWriteReceivedMessages());
+            return andesClientOutputParser.checkIfMessagesAreInOrder();
+        } else {
+            return false;
+        }
     }
 
-    public List<QueueMessageReceiver> getQueueListeners() {
-        return queueListeners;
+    /**
+     * This method returns whether received messages are transacted.
+     *
+     * @param operationOccurredIndex Index of the operated message most of the time last message.
+     * @return true if all messages are transacted, false otherwise.
+     */
+    public boolean transactedOperation(long operationOccurredIndex)
+            throws IOException {
+        if (0 < consumers.size()) {
+            AndesClientOutputParser andesClientOutputParser =
+                    new AndesClientOutputParser(consumers.get(0).getConfig()
+                                                        .getFilePathToWriteReceivedMessages());
+            return andesClientOutputParser.transactedOperations(operationOccurredIndex);
+        } else {
+            return false;
+        }
     }
 
-    public String getMessageType() {
-        return messageType;
+    /**
+     * This method returns number of duplicate received messages.
+     *
+     * @return The duplicate message count.
+     */
+    public long getTotalNumberOfDuplicates()
+            throws IOException {
+        if (0 < consumers.size()) {
+            AndesClientOutputParser andesClientOutputParser =
+                    new AndesClientOutputParser(consumers.get(0).getConfig()
+                                                        .getFilePathToWriteReceivedMessages());
+            return andesClientOutputParser.numberDuplicatedMessages();
+        } else {
+            return -1L;
+        }
     }
 
-    public void setMessageType(String messageType) {
-        this.messageType = messageType;
+    /**
+     * Sets the starting delay when starting publishers or consumers.
+     *
+     * @param startDelay The starting delay
+     */
+    public void setStartDelay(long startDelay) {
+        this.startDelay = startDelay;
     }
 
-    public int getUnsubscribeAfter() {
-        return unsubscribeAfter;
+    /**
+     * Gets the configuration file used in creating the publisher(s) or consumer(s)
+     *
+     * @return The configuration
+     */
+    public AndesJMSClientConfiguration getConfig() {
```

Comment:
If we can retrieve the configs from consumers, since we have a method to get list of consumers maybe we don't need this method.